app/components/DateInput.tsx

```diff
@@ -43,7 +50,21 @@ export function DateInput({
         value={dateToYearMonthDayHourMinuteString(date)}
         min={min ? dateToYearMonthDayHourMinuteString(min) : undefined}
         max={max ? dateToYearMonthDayHourMinuteString(max) : undefined}
-        onChange={(e) => setDate(new Date(e.target.value))}
+        onChange={(e) => {
+          const updatedDate = new Date(e.target.value);
+          setDate(updatedDate);
+
```

Comment:
One issue that already existed in this component is that there is no handling for invalid dates.
I believe most browsers will trigger an onChange with `e.target.value === ""` when the user deletes the date that we prefilled.

This then leads to `new Date("")` being put into the component state, but that's an invalid date (not throwing an error though, date objects can just be invalid).

Calling `dateToYearMonthDayHourMinuteString(date)` with an invalid date object leads to `"NaN-NaN-NaNTNaN:NaN"` 😁 

Setting the input values to these "NaN" strings will trigger a console warning by react-dom, but still "works" in that it doesn't display anything weird.

I believe we need to have the code be aware of this and test for invalid dates. One possible test you could do, to check if a date is valid is `Number.isNaN(date.getTime())`.
Where exactly is up to you, the component state could also just be the string we get from the event handler for example.

Reply:
This is fair & the initial of the focus in the PR assumed that validation was handled correctly. I can look into validating the date & coming up with a fallback when I find time later =) thanks for pointing this out!

Reply:
`Number.isNaN()` doesn't actually give me the expected result when the date object is invalid. It actually has to be the native `isNaN()` function

Reply:
Added getValidNewDateIfInvalid() function. This retrieves a valid date. If invalid, get a new Date object.
- So now, if we intentionally/accidentally delete the data in the Calendar's DateInput element, it will be reset to the current Date/Time